ape_roll/client.py

```diff
@@ -472,13 +472,27 @@ def call(self, ape_contract: ApeContractInstance, func_name, *args):
         """
         weirollContract = WeirollContract.createContract(ape_contract)
 
-        if func_name.endswith(")"):
-            # TODO: would be interesting to look at args and do this automatically
-            func = weirollContract.functionsBySignature[func_name]
+        if func_name in contract.functionsByUniqueName:
+            func = contract.functionsByUniqueName[func_name]
+        elif func_name in contract.functionsBySignature:
+            func = contract.functionsBySignature[func_name]
         else:
-            func = weirollContract.functionsByUniqueName[func_name]
+            raise ValueError(f"Unknown func_name ({func_name}) on {ape_contract}")
```

Comment:
i copied this to match delegatecall. Moving it to a helper function that both use that checks inputs vs args would be interesting, but being explicit is easy enough for now